scripts/pre_commit_linter.py

```diff
@@ -714,11 +718,144 @@ def _check_spacing(all_files):
     return summary_messages
 
 
+def _check_comments(all_files):
+    """This functions ensures that comments end in a period."""
+    print 'Starting comment checks'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    message = 'There should be a period at the end of the comment.'
+    failed = False
+    for filename in files_to_check:
+        f = open(filename, 'r')
```

Comment:
Convert this to use a `with` statement, see end of the subsection of Python [guide](https://docs.python.org/2/tutorial/inputoutput.html#methods-of-file-objects). In short, using a with statement allows for proper file closing and is better with error handling.